servers/services/src/test/java/org/projectnessie/services/rest/AbstractTestRest.java

```diff
@@ -0,0 +1,1210 @@
+import static org.projectnessie.model.Validation.REF_NAME_OR_HASH_MESSAGE;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.net.URI;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.OptionalInt;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.projectnessie.api.ContentsApi;
+import org.projectnessie.api.TreeApi;
+import org.projectnessie.client.NessieClient;
+import org.projectnessie.client.StreamingUtil;
+import org.projectnessie.client.http.HttpClient;
+import org.projectnessie.client.http.HttpClientException;
+import org.projectnessie.client.rest.NessieBadRequestException;
+import org.projectnessie.client.rest.NessieHttpResponseFilter;
+import org.projectnessie.error.NessieConflictException;
+import org.projectnessie.error.NessieNotFoundException;
+import org.projectnessie.model.Branch;
+import org.projectnessie.model.CommitMeta;
+import org.projectnessie.model.Contents;
+import org.projectnessie.model.Contents.Type;
+import org.projectnessie.model.ContentsKey;
+import org.projectnessie.model.EntriesResponse;
+import org.projectnessie.model.EntriesResponse.Entry;
+import org.projectnessie.model.Hash;
+import org.projectnessie.model.IcebergTable;
+import org.projectnessie.model.ImmutableDeltaLakeTable;
+import org.projectnessie.model.ImmutableEntry;
+import org.projectnessie.model.ImmutableHiveDatabase;
+import org.projectnessie.model.ImmutableHiveTable;
+import org.projectnessie.model.ImmutableMerge;
+import org.projectnessie.model.ImmutableOperations;
+import org.projectnessie.model.ImmutablePut;
+import org.projectnessie.model.ImmutableSqlView;
+import org.projectnessie.model.LogResponse;
+import org.projectnessie.model.MultiGetContentsRequest;
+import org.projectnessie.model.MultiGetContentsResponse.ContentsWithKey;
+import org.projectnessie.model.Operation;
+import org.projectnessie.model.Operation.Delete;
+import org.projectnessie.model.Operation.Put;
+import org.projectnessie.model.Operation.Unchanged;
+import org.projectnessie.model.Operations;
+import org.projectnessie.model.Reference;
+import org.projectnessie.model.SqlView;
+import org.projectnessie.model.SqlView.Dialect;
+import org.projectnessie.model.Tag;
+
+public abstract class AbstractTestRest {
```

Comment:
Is there anything changed in `AbstractTestRest` from `TestRest`?

Reply:
the only real difference is `protected void init(URI uri)`. everything else is just a copy/paste